core/src/main/scala/org/apache/spark/resource/ResourceInformation.scala

```diff
@@ -35,3 +41,25 @@ class ResourceInformation(
 
   override def toString: String = s"[name: ${name}, addresses: ${addresses.mkString(",")}]"
 }
+
+private[spark] object ResourceInformation {
+  /**
+   * Parses a JSON string into a [[ResourceInformation]] instance.
+   */
+  def parseJson(json: String): ResourceInformation = {
+    implicit val formats = DefaultFormats
+    try {
+      parse(json).extract[ResourceInformationJson].toResourceInformation
+    } catch {
+      case NonFatal(e) =>
+        throw new SparkException(s"Error parsing JSON into ResourceInformation:\n$json\n", e)
```

Comment:
Maybe, given tips of what is right JSON format for user ?

Reply:
Added.